src/sparse/elementwise_op.h

```diff
@@ -0,0 +1,22 @@
+/*!
+ *  Copyright (c) 2022 by Contributors
+ * \file sparse/elementwise_op.h
+ * \brief DGL C++ sparse elementwise operators
+ */
+#ifndef DGL_SPARSE_ELEMENTWISE_OP_H_
+#define DGL_SPARSE_ELEMENTWISE_OP_H_
+
+#include <torch/custom_class.h>
+
+#include "./sparse_matrix.h"
+
+namespace dgl {
+namespace sparse {
+
+c10::intrusive_ptr<SparseMatrix> Add(
+    c10::intrusive_ptr<SparseMatrix> A, c10::intrusive_ptr<SparseMatrix> B);
```

Comment:
We should have a discussion here. I do have a concern that if we only have cpp implementation, we cannot flexibly do the operator overloading as in Python. This is because when we bind a C++ function to Python, the signature is fixed. We cannot have `__add__` for both `SparseMatrix` and `Diagmatrix`.

Reply:
We can have a python function that invokes different C++ functions depending on the object types.

Reply:
I tried but I cannot override the `__add__` function of the `ScriptClass` in Python side. It will invokes its `__getattr__` function in the C++ and `__add__` is not registered in C++.

Reply:
The `SparseMatrix` we get in the `cpp_interface.py` is a `ScriptClass`. When we call `SparseMatrix()` in Python, we invoke the `__call__` function of `ScriptClass` in C++ to construct a C++ SparseMatrix. Then we cannot override the `ScriptClass` to override the method function of `SparseMatrix`.